src/library/scala/collection/mutable/BitSet.scala

```diff
@@ -107,50 +103,65 @@ class BitSet(protected[collection] final var elems: Array[Long])
     *  @param   other  the bitset to form the union with.
     *  @return  the bitset itself.
     */
-  def |= (other: BitSet): this.type = {
+  def |= (other: collection.BitSet): this.type = {
     ensureCapacity(other.nwords - 1)
-    for (i <- Range(0, other.nwords))
+    var i = 0
+    val othernwords = other.nwords
+    while (i < othernwords) {
       elems(i) = elems(i) | other.word(i)
+      i += 1
+    }
     this
   }
   /** Updates this bitset to the intersection with another bitset by performing a bitwise "and".
     *
     *  @param   other  the bitset to form the intersection with.
     *  @return  the bitset itself.
     */
-  def &= (other: BitSet): this.type = {
+  def &= (other: collection.BitSet): this.type = {
     // Different from other operations: no need to ensure capacity because
     // anything beyond the capacity is 0.  Since we use other.word which is 0
     // off the end, we also don't need to make sure we stay in bounds there.
-    for (i <- Range(0, nwords))
+    var i = 0
+    val thisnwords = nwords
+    while (i < thisnwords) {
       elems(i) = elems(i) & other.word(i)
+      i += 1
+    }
     this
   }
   /** Updates this bitset to the symmetric difference with another bitset by performing a bitwise "xor".
     *
     *  @param   other  the bitset to form the symmetric difference with.
     *  @return  the bitset itself.
     */
-  def ^= (other: BitSet): this.type = {
+  def ^= (other: collection.BitSet): this.type = {
     ensureCapacity(other.nwords - 1)
-    for (i <- Range(0, other.nwords))
+    var i = 0
+    val othernwords = other.nwords
+    while (i < othernwords) {
+
       elems(i) = elems(i) ^ other.word(i)
+      i += 1
+    }
     this
   }
   /** Updates this bitset to the difference with another bitset by performing a bitwise "and-not".
     *
     *  @param   other  the bitset to form the difference with.
     *  @return  the bitset itself.
     */
-  def &~= (other: BitSet): this.type = {
-    ensureCapacity(other.nwords - 1)
-    for (i <- Range(0, other.nwords))
+  def &~= (other: collection.BitSet): this.type = {
+    var i = 0
+    val max = Math.min(nwords, other.nwords)
+    while (i < max) {
       elems(i) = elems(i) & ~other.word(i)
+      i += 1
+    }
     this
```

Comment:
Note that we actually don't have to ensure capacity here, since were just removing any elements in `other` from `this`